pkg/ocp/machinesets_test.go

```diff
@@ -167,6 +170,49 @@ var _ = Describe("K8s MachineSetDeployer", func() {
 			})
 		})
 	})
+
+	Context("on DeleteAll", func() {
+		BeforeEach(func() {
+			machineSet.SetName(machineSetName)
+		})
+
+		When("the machine set exists", func() {
+			BeforeEach(func() {
+				machineSet.SetName(machineSetName)
+				_, err := msClient.Create(context.TODO(), machineSet, metav1.CreateOptions{})
+				Expect(err).To(Succeed())
+				machineSet.SetName(machineSetNotSubmarienrName)
+				_, err = msClient.Create(context.TODO(), machineSet, metav1.CreateOptions{})
+				Expect(err).To(Succeed())
+			})
+
+			It("should successfully delete the machine set", func() {
+				Expect(deployer.DeleteAll(machineSet, "submariner")).To(Succeed())
+
+				_, err := msClient.Get(context.TODO(), machineSetNotSubmarienrName, metav1.GetOptions{})
+				Expect(err).To(Succeed())
+
+				_, err = msClient.Get(context.TODO(), machineSetName, metav1.GetOptions{})
+				Expect(apierrors.IsNotFound(err)).To(BeTrue())
+			})
```

Comment:
I think we should separate the test cases:

```suggestion
			It("should successfully delete the matching machine set", func() {
				Expect(deployer.DeleteAll(machineSet, "submariner")).To(Succeed())
				
				_, err = msClient.Get(context.TODO(), machineSetName, metav1.GetOptions{})
				Expect(apierrors.IsNotFound(err)).To(BeTrue())
			})
			
			It("should not delete the non-matching machine set", func() {
				Expect(deployer.DeleteAll(machineSet, "submariner")).To(Succeed())

				_, err := msClient.Get(context.TODO(), machineSetNotSubmarienrName, metav1.GetOptions{})
				Expect(err).To(Succeed())
			})
```